### ISSUE TITLE
Make test failures clearer

### ISSUE THREAD

**Issue by justinsb** [GoogleCloudPlatform/k8s-config-connector#1554]
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

Our test failures can sometimes be a "bit" opaque.  It's only going to get worse when we start running all our tests in parallel (which we can do if we can get to mockkubeapiserver).

I think the ideal output would be if we had each test output written to its own file in artifacts.  (Maybe just the failures?)

I'm not entirely sure of the best way to achieve this, probably the most future proof way is to have a wrapper for go test, and we either pipe the test output to our wrapper or we run `go test` as a subprocess.  stderr is probably the distinguishing feature there.

Technically it is possible to parse the go "plaintext" output, using the `===` lines to split out sections, but it's probably easier just to run with `-json` and then write the output direct to files, and just print our own summary data at the end.  For example, in github actions, it probably suffices to just print one line when each subtest finishes.  All the output would be written to files so that it ends up in artifacts.

We could also dump the test output to stdout/stderr (only) on a failure, just so people don't have to download artifacts, but we can sort of see how that goes.

**Comment by justinsb** [comment]
/assign @anhdle-sso 

I know you've been thinking about this, so I thought I would dump my suggestions into a bug.  If you want to riff on any of these ideas, we can do it here!

**Comment by anhdle-sso** [comment]
Thanks for creating the issue. For the first improvement, I would like to reprint the failure + its logs at the end. Also, I am looking into gotestsum if they have any good feature that we can use. Ideally, separate artifacts would be best.

**Comment by justinsb** [comment]
Oh cool, `gotestsum` looks like the sort of thing I was thinking about - probably a great place to start!

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
